src/Bucket/MetricOperation.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace HelloFresh\Stats\Bucket;
+
+
+use HelloFresh\Stats\Bucket;
+
+class MetricOperation extends \SplFixedArray
+{
+    const LENGTH = 3;
+
+    /**
+     * MetricOperation constructor.
+     *
+     * @param array $operations
+     */
+    public function __construct(array $operations = [])
+    {
+        parent::__construct(static::LENGTH);
+
+        // ensure that operations is not bigger than allowed
+        array_splice($operations, static::LENGTH);
```

Comment:
I think you forgot to set the offset to 0 aka `array_splice($operations, 0, static::LENGTH);`

Reply:
```php
$a = ['bar', 'baz', 'qaz', 'qux'];
$r = array_splice($a, 0, 3);
var_dump($r);
var_dump($a);
```

results in:

```
array(3) {
  [0]=>
  string(3) "bar"
  [1]=>
  string(3) "baz"
  [2]=>
  string(3) "qaz"
}
array(1) {
  [0]=>
  string(3) "qux"
}
``

And I do not understand why, although according to function description you're correct.